infoschema/tables.go

```diff
@@ -638,32 +659,32 @@ func dataForTables(ctx sessionctx.Context, schemas []*model.DBInfo) [][]types.Da
 				Type: createTimeTp,
 			}
 			record := types.MakeDatums(
-				catalogVal,      // TABLE_CATALOG
-				schema.Name.O,   // TABLE_SCHEMA
-				table.Name.O,    // TABLE_NAME
-				"BASE TABLE",    // TABLE_TYPE
-				"InnoDB",        // ENGINE
-				uint64(10),      // VERSION
-				"Compact",       // ROW_FORMAT
-				uint64(0),       // TABLE_ROWS
-				uint64(0),       // AVG_ROW_LENGTH
-				uint64(16384),   // DATA_LENGTH
-				uint64(0),       // MAX_DATA_LENGTH
-				uint64(0),       // INDEX_LENGTH
-				uint64(0),       // DATA_FREE
-				table.AutoIncID, // AUTO_INCREMENT
-				createTime,      // CREATE_TIME
-				nil,             // UPDATE_TIME
-				nil,             // CHECK_TIME
-				table.Collate,   // TABLE_COLLATION
-				nil,             // CHECKSUM
-				"",              // CREATE_OPTIONS
-				table.Comment,   // TABLE_COMMENT
+				catalogVal,             // TABLE_CATALOG
+				schema.Name.O,          // TABLE_SCHEMA
+				table.Name.O,           // TABLE_NAME
+				"BASE TABLE",           // TABLE_TYPE
+				"InnoDB",               // ENGINE
+				uint64(10),             // VERSION
+				"Compact",              // ROW_FORMAT
+				tableRowsMap[table.ID], // TABLE_ROWS
```

Comment:
can we use statistics handle to do this:
```go
// get a statistics handle
statsHandle := domain.GetDomain(ctx).StatsHandle()

// get table row count by table ID
numTableRows := statsHandle.GetTableStats(tableID).Count
```

Reply:
Good, I'll changed it.

Reply:
@zz-jason ,
The domain package depends on infoschema, statistics and sessionctx.  It's similar that statistics depends on infoschema.  I'd also tried to return StatsHandle from session.Context, but there's still cycle dependency.  How to get tablestats without cycle importing?